### PR TITLE
Fixed input string not being in correct format

### PR DESCRIPTION
**Pull request by arxae**
Fixes some issues with strange chapter names [(like this one)](https://mangakatana.com/manga/ultra-heaven.10029)

### CODE REVIEW

File: Tranga/MangaConnectors/MangaKatana.cs [hunk 1]
```diff
@@ -154,7 +154,7 @@ public override Chapter[] GetChapters(Manga manga, string language="en")
 		//Return Chapters ordered by Chapter-Number
 		List<Chapter> chapters = ParseChaptersFromHtml(manga, requestUrl);
 		Log($"Got {chapters.Count} chapters. {manga}");
-		return chapters.OrderBy(chapter => Convert.ToSingle(chapter.chapterNumber, numberFormatDecimalPoint)).ToArray();
+		return chapters.OrderBy(chapter => chapter.chapterNumber ?? chapter.volumeNumber).ToArray();
```
Comment (C9Glax) [inline Tranga/MangaConnectors/MangaKatana.cs:157]: We don't order by volume number. Chapters are always numbered 1-x regardless of volume (so there isn't a v1c1 and v2c1)

File: Tranga/MangaConnectors/MangaKatana.cs [hunk 2]
```diff
@@ -167,9 +167,9 @@ private List<Chapter> ParseChaptersFromHtml(Manga manga, string mangaUrl)
 
 		HtmlNode chapterList = document.DocumentNode.SelectSingleNode("//div[contains(@class, 'chapters')]/table/tbody");
 
-		Regex volumeRex = new(@"Volume ([0-9]+)");
-		Regex chapterNumRex = new(@"https:\/\/mangakatana\.com\/manga\/.+\/c([0-9\.]+)");
-		Regex chapterNameRex = new(@"Chapter [0-9\.]+: (.*)");
+		Regex volumeRex = new(@"(Volume ([0-9]+))|(Vol\.[0-9]+)");
+		Regex chapterNumRex = new(@"https:\/\/mangakatana\.com\/manga\/.+\/(c([0-9\.]+))|(v[0-9]+c[0-9]+)");
+		Regex chapterNameRex = new(@"Chapter [0-9\.]+:? (.*)");
```
Comment (C9Glax) [inline Tranga/MangaConnectors/MangaKatana.cs:172]: Nice Regex, but I noticed a better way:
We can also parse the volume number from the url (like with chapters).
Would be better to just have a single Regex that can fetch both groups.
`https://mangakatana.com/manga/heavens-door-koike-keiichi.13112/c10` Chapter would be 10, Volume 1
`https://mangakatana.com/manga/ultra-heaven.10029/v3c2` Chapter would be 2, Volume 3

**Review by C9Glax — CHANGES_REQUESTED** [review]

**Comment by C9Glax** [comment]
But thank you for trying to fix things while I am working on other stuff :)

**Comment by C9Glax** [comment]
Also just putting the reference in here: #83

File: Tranga/MangaConnectors/MangaKatana.cs (hunk 1 above)
Comment (arxae) [reply Tranga/MangaConnectors/MangaKatana.cs]: I didn't seem to get the idea. Reverted this.

File: Tranga/MangaConnectors/MangaKatana.cs (hunk 2 above)
Comment (arxae) [reply Tranga/MangaConnectors/MangaKatana.cs]: Problem is that mangakatana seems to be a bit inconsistent with the namings in the url. Found at least 5 variants. But dusted off the little regex skills i have and this should cover the ones i have seen

**Comment by arxae** [comment]
Oh it seems my gitignore change snuck in as well. Hope that's fine too (it's just to exclude the .vscode folder)

**Review by C9Glax — APPROVED** [review]
Looks good!

**Pull request merged**